internal/entities/properties/service/service.go

```diff
@@ -156,9 +164,29 @@ func (ps *propertiesService) RetrieveAllProperties(
 	return refreshedProps, nil
 }
 
+// RetrieveAllPropertiesForEntity fetches a single property for the given an entity
+// for properties model. Note that properties will be updated in place.
+func (ps *propertiesService) RetrieveAllPropertiesForEntity(
+	ctx context.Context, efp *models.EntityForProperties,
```

Comment:
This is fine, but do you think that in a follow up we could change RetrieveAllProperties to accept EntityForProperties or just an Entity that alrady has projectID, providerID and type and do something like:

newProps, err := ps.RetrieveAllProperties(ctx, prov, entity, lookupProps)
efp := NewEntityFromPropertiesWithInstance(entity, newProps)